tests/ipfwd/test_dir_bcast.py

```diff
@@ -1,19 +1,106 @@
 ]
 
-def test_dir_bcast(duthosts, rand_one_dut_hostname, ptfhost, tbinfo):
+logger = logging.getLogger(__name__)
+
+PTF_TEST_PORT_MAP = '/root/ptf_test_port_map.json'
+
+def parse_mux_status(output_lines):
+    '''Parse the output of "show mux status" command
+    Args:
+        output_lines (list): The output lines of "show mux status" command
+    Returns:
+        list: A dictionary, key is interface name, value is a dictionary of fields/values
+    '''
+
+    header_line = ''
+    separation_line = ''
+    separation_line_number = 0
+    for idx, line in enumerate(output_lines):
+        if line.find('----') >= 0:
+            header_line = output_lines[idx-1]
+            separation_line = output_lines[idx]
+            separation_line_number = idx
+            break
+
+    try:
+        positions = parse_column_positions(separation_line)
+    except Exception:
+        logger.error('Possibly bad command output')
+        return {}
+
+    headers = []
+    for pos in positions:
+        header = header_line[pos[0]:pos[1]].strip().lower()
+        headers.append(header)
+
+    if not headers:
+        return {}
+
+    results = {}
+    for line in output_lines[separation_line_number+1:]:
+        mux_status = []
+        for pos in positions:
+            port_status = line[pos[0]:pos[1]].strip()
+            mux_status.append(port_status)
+
+        intf = mux_status[0]
+        results[intf] = {}
+        for idx in range(1, len(mux_status)):    # Skip the first column interface name
+            results[intf][headers[idx]] = mux_status[idx]
+
+    return results
+
+
+def get_ptf_src_ports(mg_facts):
+    ptf_src_ports = []
+    for pc in mg_facts['minigraph_portchannels']:
+        for member in mg_facts['minigraph_portchannels'][pc]['members']:
+            ptf_src_ports.append(mg_facts['minigraph_ptf_indices'][member])
+    return ptf_src_ports
+
+def get_ptf_dst_ports(duthost, mg_facts, testbed_type):
+    if "dualtor" in testbed_type:
+        mux_status_out = parse_mux_status(duthost.command("show mux status")["stdout_lines"])
+    ptf_dst_ports = []
+    for vlan in mg_facts['minigraph_vlans']:
+        for member in mg_facts['minigraph_vlans'][vlan]['members']:
+            if "dualtor" not in testbed_type:
+                ptf_dst_ports.append(mg_facts['minigraph_port_indices'][member])
+            elif mux_status_out[member]["status"].replace(",", "") == 'active':
+                ptf_dst_ports.append(mg_facts['minigraph_port_indices'][member])
+    return ptf_dst_ports
+
+def ptf_test_port_map(duthost, ptfhost, mg_facts, testbed_type):
+    ptf_test_port_map = {}
+    ptf_src_ports = get_ptf_src_ports(mg_facts)
+    ptf_dst_ports = get_ptf_dst_ports(duthost, mg_facts, testbed_type)
+    ptf_test_port_map = {
+        'ptf_src_ports': ptf_src_ports,
+        'ptf_dst_ports': ptf_dst_ports
+    }
+    ptfhost.copy(content=json.dumps(ptf_test_port_map), dest=PTF_TEST_PORT_MAP)
+
+
+@pytest.mark.xfail
```

Comment:
Please add the xfail mark to conditional mark yaml file.
Currently, dir_bcast only fails on dualtor. It should be able to pass on other topologies.

Reply:
OK